app/finders/user/filtered.rb

```diff
@@ -0,0 +1,25 @@
+class User
+  class Filtered < ApplicationFinder
+    def initialize(params = {})
+      @params = params
+      @speaker =  @params.delete(:speaker)
+    end
+
+    def call
+      User.where(conditions)
+    end
+
+    private
+
+    attr_reader :params, :speaker
+
+    def conditions
+      params.keep_if { |_, value| value == '1' }
```

Comment:
To be honest I do not like the idea of using `1` and `0` for this - `true / false` looks much better. To do so you can implement you view like this
```
<%= hidden_field_tag :something, false %>
<%= check_box_tag :something, true %>
```
This way if checkbox is checked it'll submit `true`, otherwise `false` and this method can be rewritten simply as `params.keep_if { |_, v| v }`
There's also another option - you can disable unchecked checkboxes prior to form submit which will allow us to avoid this method at all